ledger/complete/common/triequeue.go

```diff
@@ -43,13 +43,23 @@ func NewTrieQueueWithValues(capacity uint, tries []*trie.MTrie) *TrieQueue {
 	return q
 }
 
-// Push pushes trie to queue.  If queue is full, it overwrites the oldest element.
-func (q *TrieQueue) Push(t *trie.MTrie) {
+// Push pushes trie to queue and return the overwritten element, and whether any old element.
+// was ejected.
+// If queue is full, it overwrites and ejects the oldest element.
+// It return (nil, false) if no item was ejected.
+// It returns(*trie.Mtrie, true) if an old trie was ejected by the pushing of the input trie
+func (q *TrieQueue) Push(t *trie.MTrie) (*trie.MTrie, bool) {
```

Comment:
added more information  about the internal state during the state transition so that it can be reused by trieCache